src/TMxpProcessor.cpp

```diff
@@ -0,0 +1,158 @@
+ * * 6 - LOCK SECURE MODE set secure mode.  Mode remains in effect until
+ *     changed.  Secure mode becomes the new default mode.
+ * * 7 - LOCK LOCKED MODE set locked mode.  Mode remains in effect until
+ *     changed.  Locked mode becomes the new default mode."
+ */
+bool TMxpProcessor::setMode(int modeCode)
+{
+    // we really do not handle these well...
+    // MXP line modes - comments are from http://www.zuggsoft.com/zmud/mxp.htm#MXP%20Line%20TagsmMXP = true; // some servers don't negotiate, they assume!
+
+    mMXP = true;
+    switch (modeCode) {
+    case 0: // open line - only MXP commands in the "open" category are allowed.  When a newline is received from the MUD, the mode reverts back to the Default mode.  OPEN MODE starts as the Default mode until changes with one of the "lock mode" tags listed below.
+        mMXP_MODE = MXP_MODE_OPEN;
+        break;
+    case 1: // secure line (until next newline) all tags and commands in MXP are allowed within the line.  When a newline is received from the MUD, the mode reverts back to the Default mode.
+        mMXP_MODE = MXP_MODE_SECURE;
+        break;
+    case 2: // locked line (until next newline) no MXP or HTML commands are allowed in the line.  The line is not parsed for any tags at all.  This is useful for "verbatim" text output from the MUD.  When a newline is received from the MUD, the mode reverts back to the Default mode.
+        mMXP_MODE = MXP_MODE_LOCKED;
+        break;
+    case 3: //  reset (MXP 0.4 or later) - close all open tags.  Set mode to Open.  Set text color and properties to default.
+        mMxpTagBuilder.reset();
+        mMXP_MODE = mMXP_DEFAULT;
+        break;
+    case 4: // temp secure mode (MXP 0.4 or later) - set secure mode for the next tag only.  Must be immediately followed by a < character to start a tag.  Remember to set secure mode when closing the tag also.
+        mMXP_MODE = MXP_MODE_TEMP_SECURE;
+        break;
+    case 5: // lock open mode (MXP 0.4 or later) - set open mode.  Mode remains in effect until changed.  OPEN mode becomes the new default mode.
+        mMXP_DEFAULT = mMXP_MODE = MXP_MODE_OPEN;
+        break;
+    case 6: // lock secure mode (MXP 0.4 or later) - set secure mode.  Mode remains in effect until changed.  Secure mode becomes the new default mode.
+        mMXP_DEFAULT = mMXP_MODE = MXP_MODE_SECURE;
+        break;
+    case 7: // lock locked mode (MXP 0.4 or later) - set locked mode.  Mode remains in effect until changed.  Locked mode becomes the new default mode.
+        mMXP_DEFAULT = mMXP_MODE = MXP_MODE_LOCKED;
+        break;
+    default:
+        qDebug().noquote().nospace() << "TBuffer::translateToPlainText(...) INFO - Unhandled MXP control sequence CSI " << QString::number(modeCode) << " z received, Mudlet will ignore it.";
+        return false;
+    }
+
+    return true;
+}
+TMXPMode TMxpProcessor::getMode() const
+{
+    return mMXP_MODE;
+}
+bool TMxpProcessor::isEnabled() const
```

Comment:
```suggestion
bool TMxpProcessor::enabled() const
```

That makes it easier to read then:

```
if (mpHost->mMxpProcessor.enabled())
```

Same with other `is` wherever it makes better readability, for example:

```cpp
if (mpHost->mMxpClient.isInLinkMode()
```

vs


```cpp
if (mpHost->mMxpClient.inLinkMode()
```